specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/secrets.json

```diff
@@ -141,6 +150,15 @@
             "schema": {
               "$ref": "#/definitions/Secret"
             }
+          },
+          "400": {
```

Comment:
Why were these status codes explicitly added without a schema? And are they actually supposed to be treated as "success" responses (e.g. autorest would generate code that no longer raises an exception when a 400, 404 or 409 response are returned)

Reply:
The pr is to address issues related to Missing Response Code definition reported by Swagger KPIs team. I believed we have to add the status codes explicitly to fix that. I tried to add the schema like "SecretResponseError" before. But some SDK generations cause failure because of response type change from <Secret> to <Object>. Any suggestion if we still need to treat this as "error" responses?

Reply:
Adding a schema will cause the following error in .Net SDK generation. The result typed is changed from Vault to Object or other error schema: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Microsoft.Azure.Management.KeyVault/src/Generated/VaultsOperations.cs#L71
`error CS0029: Cannot implicitly convert type 'Microsoft.Rest.Azure.CloudError' to 'Microsoft.Azure.Management.KeyVault.Models.Vault'`

For the "success" response concern, can it be fix if I add the x-ms-error-response?
`"x-ms-error-response": true`

Reply:
The best answer is to add a `"default"` section to the responses rather than listing individual status codes.

Reply:
We have got a report to fix swagger quality issue in s360, in which it asked us to fix [Invalid Response Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/Semantic-and-Model-Violations-Reference.md#INVALID_RESPONSE_CODE). Using "default" may not be able to fix the issue reported from s360.

Reply:
It should be enough to add a `default`.